src/__tests__/client.ts

```diff
@@ -2690,845 +2504,758 @@ describe("client", () => {
-
-          // The error arrived, run a refetch to get the third result
-          // which should now contain valid data.
-          setTimeout(() => {
-            observable.refetch().catch(() => {
-              reject("Expected good data on second refetch.");
-            });
-          }, 0);
-        },
-      };
-
-      subscription = observable.subscribe(observerOptions);
-    }
-  );
-
-  itAsync("should throw a GraphQL error", (resolve, reject) => {
-    const query = gql`
-      query {
-        posts {
-          foo
-          __typename
         }
       }
     `;
-    const errors: GraphQLError[] = [
-      new GraphQLError('Cannot query field "foo" on type "Post".'),
-    ];
-    const link = mockSingleLink({
-      request: { query },
-      result: { errors },
-    }).setOnError(reject);
+
+    const data = { allPeople: { people: [{ name: "Luke Skywalker" }] } };
+    const dataTwo = { allPeople: { people: [{ name: "Princess Leia" }] } };
+    const link = mockSingleLink(
+      { request: { query }, result: { data } },
+      { request: { query }, error: new Error("This is an error!") },
+      { request: { query }, result: { data: dataTwo } }
+    );
     const client = new ApolloClient({
       link,
-      cache: new InMemoryCache(),
+      cache: new InMemoryCache({ addTypename: false }),
     });
 
-    return client
-      .query({ query })
-      .catch((err) => {
-        expect(err.message).toBe('Cannot query field "foo" on type "Post".');
-      })
-      .then(resolve, reject);
-  });
+    const observable = client.watchQuery({
+      query,
+      notifyOnNetworkStatusChange: true,
+    });
 
-  it("should warn if server returns wrong data", async () => {
-    using _consoleSpies = spyOnConsole.takeSnapshots("error");
-    await new Promise((resolve, reject) => {
-      const query = gql`
-        query {
-          todos {
-            id
-            name
-            description
-            __typename
-          }
-        }
-      `;
-      const result = {
-        data: {
-          todos: [
-            {
-              id: "1",
-              name: "Todo 1",
-              price: 100,
-              __typename: "Todo",
-            },
-          ],
-        },
-      };
+    let stream = new ObservableStream(observable);
 
-      const link = mockSingleLink({
-        request: { query },
-        result,
-      }).setOnError(reject);
-      const client = new ApolloClient({
-        link,
-        cache: new InMemoryCache({
-          // Passing an empty map enables the warning:
-          possibleTypes: {},
-        }),
-      });
+    await expect(stream).toEmitValue({
+      loading: false,
+      networkStatus: NetworkStatus.ready,
+      data,
+    });
 
-      return client
-        .query({ query })
-        .then(({ data }) => {
-          expect(data).toEqual(result.data);
-        })
-        .then(resolve, reject);
+    await wait(0);
+    expect(observable.refetch()).rejects.toThrow();
```

Comment:
```suggestion
   await expect(observable.refetch()).rejects.toThrow();
```

Do you have ESLint enabled? That should flag this.

Reply:
Yep I do. I think I updated this file though before we got that rule in, so I'll go check for these warnings.